src/modules/thirdParty/sagas.ts

```diff
@@ -237,9 +246,24 @@ export function* thirdPartySaga(builder: BuilderAPI, catalyst: CatalystClient) {
         throw new Error('Invalid Identity')
       }
 
+      yield put(
+        openModal('ApprovalFlowModal', {
+          view: ApprovalFlowModalView.DEPLOYING_TP
+        })
+      )
+
       const promisesOfItemsBeingDeployed: (() => Promise<ItemCuration>)[] = items.map((item: Item) => async () => {
         const entity: DeploymentPreparationData = await buildTPItemEntity(catalyst, builder, collection, item, tree, hashes[item.id])
-        await catalyst.deployEntity({ ...entity, authChain: Authenticator.signPayload(identity, entity.entityId) })
+        try {
+          await catalyst.deployEntity({ ...entity, authChain: Authenticator.signPayload(identity!, entity.entityId) })
```

Comment:
I have three questions about this try catch:
- I've seen that when a promise is rejected in the queue, it throws, reaching the catch part of the sagas directly. Is it needed to have a try catch?
- One of the main problems when deploying entities is the creation of the `DeploymentPreparationData`. It often occurs that there are files that weren't uploaded correctly. Would you mind creating a custom error for the process of requesting the contents (that is done inside of the buildTIPItemEntity) so we can expose which file and for which item the deployment preparation failed?
- Can improve the errors here by including which item failed to be deployed?

Reply:
logic updated!